e2eTests/e2eTestHelper.psm1

```diff
@@ -33,10 +33,21 @@ function SetTokenAndRepository {
         invoke-git config --global core.autocrlf false
     }
 
-    if ($appKey -and $appId) {
+    if (-not $github) {
+        # Running locally - Ensure the user is authenticated with the GitHub CLI.
+        # This is required for local runs to perform GitHub-related operations.
+        invoke-gh auth status
+    } elseif ($appKey -and $appId) {
+        # Running in GitHub Actions
         $token = @{ "GitHubAppClientId" = $appId; "PrivateKey" = ($appKey -join '') } | ConvertTo-Json -Compress -Depth 99
+    } else {
+        throw "GitHub App ID and Private Key not set. In order to run end to end tests, you need a Secret called E2E_PRIVATE_KEY and a variable called E2E_APP_ID."

```

Comment:
Message could be better in case of a local run.

Reply:
Local runs (non-GitHub) would enter the first if statement and fail there with "You are not logged into any GitHub hosts. To log in, run: gh auth login"